Model/Meta/Account/Settings/Currencies/Builder.php

```diff
@@ -143,4 +130,131 @@ public function build(Store $store)
 
         return $currencies;
     }
+
+    /**
+     * Returns the decimal precision used by the currency
+     *
+     * @param $priceFormat
+     * @return bool|int
+     */
+    private function getDecimalPrecision($priceFormat)
+    {
+        $precision = 0;
+        if (($decimalPos = strpos($priceFormat, '.')) !== false) {
+            $precision = (strlen($priceFormat) - (strrpos($priceFormat, '.') + 1));
+        } else {
+            $decimalPos = strlen($priceFormat);
+        }
+        $decimalFormat = substr($priceFormat, $decimalPos);
+        if (($pos = strpos($decimalFormat, '#')) !== false) {
+            $precision = strlen($decimalFormat) - $pos - $precision;
+        }
+        return $precision;
+    }
+
+    /**
+     * Returns true if currency is defined to have no decimal part
+     * according to ISO-4217
+     *
+     * @param $currencyCode
+     * @return bool
+     */
+    private function isZeroDecimalCurrency($currencyCode)
+    {
+        foreach (self::$zeroDecimalCurrencyCodes as $zeroDecimalCurrency) {
```

Comment:
Why not `in_array `?

Reply:
Slowness, but dismissible in this scenario. Fixed, thanks.